src/core/alloc_type.hpp

```diff
@@ -41,31 +35,22 @@ class ReusableBuffer {
 	 */
 	T *Allocate(size_t count)
 	{
-		if (this->count < count) {
-			free(this->buffer);
-			this->buffer = MallocT<T>(count);
-			this->count = count;
-		}
-		return this->buffer;
+		if (this->buffer.size() < count) this->buffer.resize(count);
+		return this->buffer.data();
 	}
 
 	/**
-	 * Get buffer of at least count times T with zeroed memory.
+	 * Get buffer of at least count times T of default initialised elements.
 	 * @note the buffer might be bigger
 	 * @note calling this function invalidates any previous buffers given
 	 * @param count the minimum buffer size
 	 * @return the buffer
 	 */
 	T *ZeroAllocate(size_t count)
 	{
-		if (this->count < count) {
-			free(this->buffer);
-			this->buffer = CallocT<T>(count);
-			this->count = count;
-		} else {
-			memset(this->buffer, 0, sizeof(T) * count);
-		}
-		return this->buffer;
+		this->buffer.resize(0);
```

Comment:
```suggestion
		this->buffer.clear();
```

maybe? Not sure there is a performance impact or what-ever.

Reply:
Good point. I haven't even thought about `clear()`.